Dev/kk/book 21 modfied client for user

Sadymix opened the pull request.

Commit:
implemented multiple clients

Commit:
implemented multiple DtoLists for wrapping a list of dto objects

Commit:
implemented multiple dto class

Commit:
implemented test case - getReservationWithPassengersAndFlight for ReservationClient

Commit:
removed redundant code get methods don't have a request

Commit:
modified test case testGetSingleUser

Sadymix committed:
implemented changes for all test cases in UserClientTest

Sadymix committed:
removed unnecessary blank spaces

Commit:
removed redundant wrapper classes

Commit:
changed the implementation of multiple methods in different clients

Commit:
implemented smoke-test for all clients

Commit:
modified the test cases for UserClient

Commit:
modified testGetSingleUser testCase

Commit:
added prefix test for each test case, implemented all test cases fpr ReservationClient

Commit:
removed logging for RestTemplate and added urls for clients

Commit:
implemented testcases for PassengerClient

Commit:
removed redundant mock

Commit:
added prefixes for all test cases

Commit:
implemented test cases for AirportClient

Commit:
implemented test cases for FlightClient

Commit:
implemented test case for WeatherClient

Commit:
minor change

Commit:
minor changes

Sadymix committed:
implemented nested class - getListOfReservationDtoTestCases for 3 test cases

pszerszen commented on src/main/java/com/pgs/client/component/Client.java:
```diff
@@ -28,18 +33,89 @@
             .enabled(true)
             .roles(List.of("USER"))
             .build();
+    private static PassengerDto passengerDto = PassengerDto.builder()
```
all the private static fields are not modified, therefore can be final

pszerszen commented on src/main/java/com/pgs/client/component/Client.java:
```diff
@@ -28,18 +33,89 @@
-        log.info("Activated User: {}", activateUser);
-        var setRolesUser = userClient.setUserRoles(1L, List.of("USER", "STAFF"));
-        log.info("Set Roles User: {}", setRolesUser);
+        userClientPostConstruct();
+        reservationClientPostConstruct();
+        passengerClientPostConstruct();
+        airportClientPostConstruct();
+        flightClientPostConstruct();
```
more accurate names wold be userClientDemo() etc.

pszerszen commented on src/main/java/com/pgs/client/component/Client.java:
```diff
@@ -28,18 +33,89 @@
+        passengerClientPostConstruct();
+        airportClientPostConstruct();
+        flightClientPostConstruct();
+        weatherClient.getWeatherByCity("Warsaw");
```
please ignore the weather client in this project

pszerszen commented on src/main/java/com/pgs/client/dto/WeatherDto.java:
```diff
@@ -0,0 +1,15 @@
+@Builder
+@Data
+@AllArgsConstructor
+public class WeatherDto {
```
to be removed

pszerszen commented on src/main/java/com/pgs/client/service/AirportClient.java:
```diff
@@ -0,0 +1,71 @@
+
+    private static HttpHeaders getHeaders() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.setAccept(List.of(MediaType.APPLICATION_JSON));
```
This is repeating in all the clients. You can use [RestTemplateBuilder method](https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/web/client/RestTemplateBuilder.html#defaultHeader-java.lang.String-java.lang.String...-) for such stuff.

pszerszen commented on src/main/java/com/pgs/client/service/WeatherClient.java:
```diff
@@ -0,0 +1,23 @@
+
+@Service
+@RequiredArgsConstructor
+public class WeatherClient {
```
to be removed

pszerszen commented on src/main/resources/application.yml:
```diff
@@ -7,17 +7,15 @@ app:
+    passengers: ${app.url.api}/passengers
+    reservations: ${app.url.api}/reservations
     user: ${app.url.api}/users
+    weather: ${app.url.api}/weather
```
```suggestion
```

pszerszen commented on src/test/java/com/pgs/client/service/AirportClientTest.java:
```diff
@@ -0,0 +1,108 @@
+    @InjectMocks
+    private AirportClient airportClient;
+
+    private static final String URL = "http://localhost:8080/api/airports";
+    private static final AirportDto AIRPORT = AirportDto.builder()
```
static fields are usually above non-static (fix it along all the tests)

pszerszen commented on src/test/java/com/pgs/client/service/ReservationClientTest.java:
```diff
@@ -0,0 +1,143 @@
+    }
+
+    @Nested
+    class getListOfReservationDtoTestCases {
```
Name of classes should start with big letter

Commit:
removed weatherClient, modified fields to final in Client

Commit:
implemented headers via RestTemplateBuilder to RestTemplate bean,
removed getHeaders method from each Client,
implemented changes in test cases in UserClient and ReservationClient

Commit:
fixed the order of fields in tests

Commit:
fixed the order of fields in Client
removed weather url - redundant for application.yml

Commit:
Merge remote-tracking branch 'origin/dev/kk/BOOK-21-modfied-client-for-user' into dev/kk/BOOK-21-implement-rest-of-the-clients

# Conflicts:
#	src/test/java/com/pgs/client/service/ReservationClientTest.java

Commit:
modified ReservationClientTest
